flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/AbstractDeploymentObserver.java

```diff
@@ -199,11 +199,7 @@ protected void observeJmDeployment(
     }
 
     private Optional<Deployment> getSecondaryResource(FlinkDeployment flinkApp, Context context) {
-        return context.getSecondaryResource(
-                Deployment.class,
-                configManager.getOperatorConfiguration().getWatchedNamespaces().size() > 1
-                        ? flinkApp.getMetadata().getNamespace()
-                        : null);
+        return context.getSecondaryResource(Deployment.class);
```

Comment:
unused `flinkApp`, since it is a one line method call, do we still need a method ?

Reply:
No, I removed it most of the places.